obsidian.el

```diff
@@ -373,36 +466,53 @@ lower and upper case versions of the tags."
          (-map (lambda (s) (s-concat "#" s)))
          -distinct)))
 
-(defun obsidian-tags-backend (command &rest arg)
-  "Completion backend for company used by obsidian.el.
-Argument COMMAND company command.
-Optional argument ARG word to complete."
-  (interactive (if (and (featurep 'company)
-                        (fboundp 'company-begin-backend))
-                   (company-begin-backend 'obsidian-tags-backend)
-                 (error "Company not installed")))
-  (cl-case command
-
-    (prefix (when (and
-                   (-contains-p local-minor-modes 'obsidian-mode)
-                   (looking-back obsidian--tag-regex nil))
-              (match-string 0)))
-    (candidates (->> obsidian--tags-list
-                     obsidian-prepare-tags-list
-                     (-filter (lambda (s) (s-starts-with-p (car arg) s)))))))
-
 (defun obsidian-enable-minor-mode ()
   "Check if current buffer is an `obsidian-file-p' and toggle `obsidian-mode'."
   (and (derived-mode-p 'markdown-mode)
        (obsidian-file-p)
        (obsidian-mode t)))
 
+(defun obsidian--find-all-files()
+  "Return a list of all obsidian files in the vault."
+  (let ((all-files (directory-files-recursively obsidian-directory "\.*$")))
+    (-filter #'obsidian-file-p all-files)))
+
+(defun obsidian-clear-cache ()

```

Comment:
`obsidian-update` is now a much more lightweight function; `obsidian-clear-cache` can be used to force a cache purge